tcms/testcases/models.py

```diff
@@ -303,6 +303,14 @@ class BugSystem(models.Model, UrlMixin):
         max_length=256, null=True, blank=True, verbose_name="API password or token"
     )
 
+    api_access_key = models.CharField(
+        max_length=256,
+        null=True,
+        blank=True,
+        verbose_name="API access key",
+        help_text="Optional API token or access key used for authentication. Takes precedence over username/password if provided.",
+    )
```

Comment:
This field isn't really necessary. The existing `api_password` field can be used instead and the code will be greatly simplified.


In addition `self.rpc_credentials()` must always be used to fetch login credentials (see methods you've already altered above) b/c it provides the ability to override them with a user defined function - adding a 3rd field just doesn't fit into the existing architecture.

Reply:
Done, updated as requested.

Reply:
Done, updated as requested.